prometheus/vec.go

```diff
@@ -216,7 +228,7 @@ func (m *MetricVec) hashLabelValues(vals []string) (uint64, error) {
 	}
 	h := hashNew()
 	for _, val := range vals {
-		h = hashAdd(h, val)
+		h = m.hashAdd(h, val)
```

Comment:
This is about hashing, not about tuples. Only add this character while creating the hash, nowhere else.

https://github.com/prometheus/common/blob/master/model/signature.go